src/main/scala/higherkindness/rules_scala/common/sbt-testing/JUnitXmlReporter.scala

```diff
@@ -8,8 +8,11 @@ import scala.collection.mutable.ListBuffer
 import scala.xml.{Elem, XML}
 
 class JUnitXmlReporter(tasksAndEvents: ListBuffer[(String, ListBuffer[Event])]) {
-  private def escape(str: String): String =
-    str.replaceAll("&", "&amp;").replaceAll("\"", "&quot;").replaceAll("<", "&lt;").replaceAll(">", "&gt;");
+  private def escape(info: Any): String = info match {
+    case str: String =>
+      str.replaceAll("&", "&amp;").replaceAll("\"", "&quot;").replaceAll("<", "&lt;").replaceAll(">", "&gt;")
+    case _ => ""
```

Comment:
Does this prevent the NPE, but still cause problems in the application?

Reply:
I am not sure what do you mean by causing problems in the application?

Reply:
How does the application handle the empty string?

As in, did you convert the NPE into really weird runtime behavior?

Reply:
The XML reporter is the final step of the application, the only purpose is to output an XML file to Bazel output directory. The empty string will only be used in the XML file, and I don't think we are passing that XML to any other application except uploading it to Jenkins. Does it address your concern?

Reply:
Provided Jenkins and anything else that uses it still works, then yes.

Reply:
I will test it on Jenkins now.